src/frontend/js/components/PaymentButton/index.spec.tsx

```diff
@@ -467,60 +530,84 @@ describe('PaymentButton', () => {
     expect($button.disabled).toBe(false);
 
     // - User clicks on pay button
+    // fetchMock.get(
+    //   `https://joanie.test/api/v1.0/orders/?course=00000&product=${product.id}&state=pending&state=validated&state=submitted`,
+    //   [
+    //     {
+    //       ...order,
+    //       state: OrderState.SUBMITTED,
+    //     },
+    //   ],
+    //   { overwriteRoutes: true },
+    // );
+    console.log(
+      'calls',
+      fetchMock.calls().map((call) => call[0]),
+    );
     await act(async () => {
       fireEvent.click($button);
     });
 
-    // - Route to create order should have been called
-    // - Furthermore, as payment succeeded immediately, order should have been refetched
-    expect(fetchMock.calls()).toHaveLength(3);
-    expect(fetchMock.calls()[0][0]).toBe('https://joanie.test/api/v1.0/orders/');
-    expect(JSON.parse(fetchMock.calls()[0][1]!.body as string)).toEqual({
-      course: '00000',
-      product: product.id,
-    });
-
-    expect(fetchMock.calls()[1][0]).toBe(`https://joanie.test/api/v1.0/orders/${order.id}/submit/`);
-    expect(JSON.parse(fetchMock.calls()[1][1]!.body as string)).toEqual({
-      billing_address: billingAddress,
-      credit_card_id: creditCard.id,
-    });
-
-    expect(fetchMock.calls()[2][0]).toBe(`https://joanie.test/api/v1.0/orders/${order.id}/`);
-
-    // - Spinner should be displayed
-    screen.getByText('Payment in progress');
-
-    // - Payment interface should not be displayed
-    expect(screen.queryByText('Payment interface component')).toBeNull();
-
-    fetchMock.resetHistory();
-    // - Wait until order has been polled 29 times.
-    await jest.advanceTimersToNextTimerAsync(PAYMENT_SETTINGS.pollLimit);
-    expect(fetchMock.calls()).toHaveLength(PAYMENT_SETTINGS.pollLimit - 1);
-
-    // - This round should be the last after which the order should be aborted
-    await act(async () => {
-      jest.runOnlyPendingTimers();
-    });
-
-    await waitFor(
-      async () => {
-        expect(fetchMock.calls()).toHaveLength(PAYMENT_SETTINGS.pollLimit);
-        expect(fetchMock.lastUrl()).toBe(`https://joanie.test/api/v1.0/orders/${order.id}/abort/`);
-      },
-      {
-        timeout: 1100,
-      },
-    );
-
-    expect(JSON.parse(fetchMock.lastOptions()!.body!.toString())).toEqual({
-      payment_id: paymentInfo.payment_id,
-    });
-
-    // - An error message should be displayed and focused (for screen reader users)
-    const $error = screen.getByText('An error occurred during payment. Please retry later.');
-    expect(document.activeElement).toBe($error);
+    // // - Route to create order should have been called
```

Comment:
There is an error that i don't understand: 
```sh
Unmatched POST to https://joanie.test/api/v1.0/orders/1cc7077a-b494-47b6-9f43-b5dab3ba6f98/abort/
```
where this url is mock on line 508.